File: back-end/src/main/java/es/ull/project/application/service/workflow/WorkFlowCreateService.java
package es.ull.project.application.service.workflow;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;

import es.ull.project.application.repository.WorkFlowRepository;
import es.ull.project.application.usecase.workflow.WorkFlowCreateUseCase;
import es.ull.project.domain.entity.WorkFlow;
import es.ull.project.domain.valueobject.WorkFlowName;
import es.ull.project.domain.valueobject.Node;

public class WorkFlowCreateService implements WorkFlowCreateUseCase {

    @Autowired
    private WorkFlowRepository repository;

    @Override
    public WorkFlow createWorkFlow(WorkFlowName name, List<Node> nodes, boolean enabled) {
        final WorkFlow workflow = new WorkFlow(name, nodes, enabled);
        return repository.save(workflow);
    }
}

File: front-end/src/application/service/workflow/workflow-create-service.ts
import { WorkFlowRepository } from '@/application/repository/workflow-repository';
import { WorkFlowCreateUseCase } from '@/application/usecase/workflow/workflow-create-use-case';
import { WorkFlow } from '@/domain/entity/workflow';
import { Node } from '@/domain/value-object/node';
import { WorkFlowName } from '@/domain/value-object/workflow-name';
import { Either } from '@ull-tfg/ull-tfg-typescript';
import type { DataError } from '@ull-tfg/ull-tfg-typescript';


export class WorkFlowCreateService implements WorkFlowCreateUseCase {

  constructor(private readonly flowRepository: WorkFlowRepository) { }

  async createWorkFlow(
    workFlowName: WorkFlowName,
    nodes: Node[],
    creationDate: Date | string,
    enabled: boolean
  ): Promise<Either<DataError, WorkFlow>> {
    
    const workflow = new WorkFlow(workFlowName, nodes, enabled, creationDate);
    return this.flowRepository.saveWorkFlow(workflow);
  }
}
